### DISCUSSION TITLE
Installation Instructions

### DISCUSSION THREAD

**Discussion by VeissieV** [anapnoe/stable-diffusion-webui-ux#27] [Q&A] [answered]
Edit: weird... All I did was change my script to this and the UX started showing up.
Conda activate automatic1111
Cd X:\GAN\A1111_stable-diffusion-webui\A1111-UX
cd extensions
ls -Attributes D | foreach{cd $_;git pull;cd ..}
cd ..
git switch master
git pull
python launch.py --opt-sdp-attention --opt-sdp-no-mem-attention --always-batch-cond-uncond

But I'm still wondering why there would be a necessity to clone the A1111 repo if everything needed is in the UX repo. Maybe I'm still accessing some dependency that the A1 repo included somehow..

Hello, I'm having an issue getting the UX working again with a fresh install in hopes of remedying some issues or at least confirming nothing from my original A1111 install was messing with the UX install. Initially I copied my A1111 installation, then copied everything from this UX clone into that folder replacing files when prompted. This worked as far as getting the UX up and running. I then made some modifications to it to my taste and have been using it fine for a few days now and loving it. 

I noticed a few issues like dropping in png chunks in the prompt field no longer working, code appearing a the bottom of the UX randomly(as long as the UX or longer), and a lot of stacked image symbols in the CivitAI extension before and after the model images and questioned my install overall, especially in regard to extensions and thought it might be good to just do a fresh install and then copy in my modifications since I originally started from my A1 install that had all my extensions and settings. Maybe this issue with the CivitAI extension is known or isn't just on my end..

After cloning this repo and running it from my launch script the standard A1 UI appears and works. It is for sure launching from the files I cloned from this repo as there are no extensions or other models(other than SD1.5), etc. I don't quite understand how the UX is supposed to be applied or why the instructions say to clone the A1 repo first if I didn't do that with this fresh install and have the standard UI from only cloning the UX repo. Is there something in the standard A1 repo required to make the UX work and could that be included in this repo if the standard UI works without the A1 repo? The instructions don't say anything about the UX files, just the standard A1 install instructions. Maybe I missed it..

I'm launching it with a script that activates my environment, changes the directory to this repos location, then uses the launch.py instead of the webui-user.bat since I'm sharing the Torch2 environment with all my A1 applications and using the webui-user.bat would create a new venv. I'm also symlinking common folders such as models and outputs so I can share those between applications as well.(though the symlinking hasn't been done for this fresh install yet)

I'm using Fork to manage repos and would think I could just checkout new versions and choose don't change local changes whenever there's a new update making the update process easy.

So to summarize the regular A1 UI is working now and I have it setup in this way that would seem ideal for updating without changing local changes from Fork but the UX just isn't triggering while the standard UI is and works without cloning the A1111 repo. 

Hopefully this wasn't too confusing...

**Comment by anapnoe** [comment, marked as answer]
The UIUX is a fork of the original A1111 project you don't need to have the A1111 installed to apply the UIUX
Just download the project and copy your models and your extensions from your other installation 
now since the last commit the original A1111 became unstable many changes and functionality was broken due to gradios framework upgrade. UIUX doesn't want to update to the last commit until things sort out so if you downloaded the latest unstable version from the original repository A1111 but you want to run UIUX you need to revert the process as mentioned first you install UIUX then you copy the models and extensions if something doesn't work disable all non inbuild extensions and enable them one by one to find what causes the problem many of the extension developers try to patch things out as we speak so their extension will run on the new unstable version which has its own issues to solve there is no versioning for extensions if the new one doesn't work you need to download an older one

**Comment by mart-hill** [comment]
I've noticed, that the installation instructions on the front page show, how to install AUTOMATIC's WebUI, but not your fork... Shouldn't there be `git clone https://github.com/anapnoe/stable-diffusion-webui-ux.git`? 🙂

**Reply by anapnoe** [reply]
yes it should. Initially my hope was to merge the UI UX with the main branch but it would be nice to fix the front page and maybe create some video showing the features any people who want to help are welcome :)
